### PR TITLE
SGSSB: Guarantee a user's staffmon

### PR DESCRIPTION
**Pull request by DeathlyPlays**
This PR will make a user gain their staffmon if their username matches the staffmons username

### CODE REVIEW

**Comment by DeathlyPlays** [comment]
Ill try to fix this when I'm home

**Comment by DeathlyPlays** [comment]
Works now

File: mods/sgssb/scripts.js [hunk 1]
```diff
@@ -410,49 +411,41 @@ exports.BattleScripts = {
...
+				let monName;
+				for (let mon in sets) {
+					if (toId(mon) === userid) {
+						monName = mon;
+						break;
+					}
+				}
+				if (monIds.indexOf(userid) === -1 && monName) {
+					pool[2] = monName;
+				}
```
Comment (HoeenCoder) [inline mods/sgssb/scripts.js]: This is unessecarly complicated.

Do:
```javascript
let monIds = pool.slice(0, 6).map(function (p) {
    return toId(p);
});

for (let mon in sets) {
    if (toId(mon) === userid && monIds.indexOf(userid) === -1) {
        pool[1] = mon;
        break;
    }
}
```
(dont CP it, travis will cry)

File: mods/sgssb/scripts.js [hunk 2]
```diff
@@ -410,49 +411,41 @@ exports.BattleScripts = {
...
-					hp: 31,
-					atk: 31,
-					def: 31,
-					spa: 31,
-					spd: 31,
-					spe: 31,
-				}) {
-					set.ivs[iv] = iv in set.ivs ? set.ivs[iv] : 31;
+				for (let iv in {hp:31, atk:31, def:31, spa:31, spd:31, spe:31}) {
+					set.ivs[iv] = set.ivs[iv] ? set.ivs[iv] : 31;
```
Comment (HoeenCoder) [inline mods/sgssb/scripts.js]: Handle an IV of 0.

File: mods/sgssb/scripts.js [hunk 3]
```diff
@@ -410,49 +411,41 @@ exports.BattleScripts = {
...
-				set.evs = {
-					hp: 84,
-					atk: 84,
-					def: 84,
-					spa: 84,
-					spd: 84,
-					spe: 84,
-				};
-			}
+			// Assuming the hardcoded set evs are all legal. LOLOLOLOLOL
```
Comment (HoeenCoder) [inline mods/sgssb/scripts.js]: No need for the lol

Comment (DeathlyPlays) [reply mods/sgssb/scripts.js]: yes there is (jk)

**Pull request merged**